monai/bundle/scripts.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Optional, Sequence, Union
+
+from monai.bundle.config_parser import ConfigParser
+from monai.bundle.config_reader import ConfigReader
+from monai.bundle.utils import update_default_args
+
+
+def run(
+    meta_file: Optional[Union[str, Sequence[str]]] = None,
+    config_file: Optional[Union[str, Sequence[str]]] = None,
+    override: Optional[Union[Dict, str]] = None,
+    target: Optional[str] = None,
+    args_file: Optional[str] = None,
+    **kwargs,
+):
+    """
+    Specify metadata file and config file to run a regular training or evaluation program.
+    It's used to execute most of the supervised training, evaluation or inference cases.
+
+    Typical usage examples:
+
+    .. code-block:: bash
+
+        # Execute this module as CLI entry:
+        python -m monai.bundle run --meta_file=<meta path> --config_file=<config path> --target=trainer
+
+        # Override some config values at runtime, set `override` as a dict:
+        python -m monai.bundle run --override='{"net#<args>#ndims": 2}' ...
+
+        # Override some config values at runtime:
+        python -m monai.bundle run --"net#<args>#input_chns" 1 ...
```

Comment:
Hi @Nic-Ma, I just added `**kwargs` to support this use case of `python -m monai.bundle run --"net#<args>#input_chns" 1`, do you have any concerns? also I'm not sure about `override: Optional[Union[Dict, str]]`, when is `str` input useful?

Reply:
`override` should not be `str`, that's outdated code and I forgot to update the typehints.

Thanks.

Reply:
Hi @wyli ,

I understand your proposal. But I feel it's a little bit confusing to have both `override` and `kwargs` args, maybe we can simply keep only one of them?
If only keeps `kwargs`, it can't support the case that the overriding keys happen to be "target", "config_file", "meta_file" or "args_file". And we may have more similar issues in other future scripts.
And may also consider supporting `kwargs` in the `args_file` file?
What do you think?

Thanks in advance.

Reply:
I reviewed the usage and found that providing a dict input at CLI is not that straightforward in terms of syntax '{"net#<args>#ndims": 2}'. The kwargs would make it simpler, I think we should keep it. Perhaps we remove 'override'? Please help look into this,(sometimes it's not override the existing options but adding new options?)

Reply:
OK, so I think we need to change `target` arg to other names like `runner_name`, `target` is slightly easier to conflict with some keys in the config content.

Thanks.